src/components/extensions/CustomLink.js

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+/** @jsx jsx */
+import { jsx, css } from "@emotion/core";
+import { LinkIcon } from "../core/icons";
+import Link from "next/link";
+import styled from "@emotion/styled";
+
+const CodeLink = styled.a`
+  text-decoration: none;
+  svg {
+    vertical-align: middle;
+    margin-left: 2px;
+    margin-right: 4px;
+    color: var(--code-inline-color);
+  }
+  &:hover,
+  &:active {
+    text-decoration: none;
+    code {
+      background-color: #efe8e8;
+    }
+  }
+`;
+
+const CustomLink = ({ children, ...props }) => {
```

Comment:
1. Choose different function name. Word custom isn't every informative
2. Can you import every props separately? Just like you did it in your previous PR. We would also like to add `PropTypes` to this component.